packages/api/src/webhooks/cannon.ts

```diff
@@ -200,12 +218,19 @@ export default class WebhookCannon {
     return newInterval | 0;
   };
 
-  retry(trigger: messages.WebhookTrigger) {
+  retry(trigger: messages.WebhookTrigger, webhookPayload?: any, err?: any) {
     if (trigger?.retries >= MAX_RETRIES) {
       console.log(
         `Webhook Cannon| Max Retries Reached, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
       );
-      return;
+      try {
+        this.notifyFailedWebhook(trigger, webhookPayload, err);
+        return;
+      } catch (err) {
+        console.error(
+          `Webhook Cannon| Error sending notification email to user, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
+        );
+      }
```

Comment:
The return should be after the `catch` block otherwise when an exception is thrown sending the email we are going to retry the webhook past the maximum limit we want to allow. 

```suggestion
      try {
        this.notifyFailedWebhook(trigger, webhookPayload, err);
      } catch (err) {
        console.error(
          `Webhook Cannon| Error sending notification email to user, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
        );
      }
      return;
```

Reply:
Nice catch!